src/protocol/vm/state.rs

```diff
@@ -203,7 +204,9 @@ impl VMPoolState<PreCachedDB> {
                 } else {
                     let code =
                         Bytecode::new_raw(Bytes::from(bytecode.clone().ok_or_else(|| {
-                            TychoSimulationError::DecodingError("Expected bytecode".into())
+                            TychoSimulationError::from(VMError::DecodingError(
+                                "Expected bytecode".into(),
```

Comment:
Usually when we use a `.expect` we phrase the error message as what was expected. Here when we raise an error we phrase the message as what went wrong. In this case it looks like we're raising because the bytecode is None?